packages/hopli/src/migrate_safe_module.rs

```diff
@@ -0,0 +1,161 @@
+use crate::{
+    identity_input::LocalIdentityArgs,
+    key_pair::read_identities,
+    password::PasswordArgs,
+    process::{set_process_path_env, child_process_call_foundry_migrate_safe_module},
+    utils::{Cmd, HelperErrors},
+};
+use clap::Parser;
+use core_crypto::keypairs::Keypair;
+use core_crypto::types::ToChecksum;
+use ethers::{
+    types::U256,
+    utils::parse_units, //, types::U256, utils::format_units, ParseUnits
+};
+use log::{log, Level};
+use std::{env, str::FromStr};
+use utils_types::primitives::Address;
+
+/// CLI arguments for `hopli migrate-safe-module`
+#[derive(Parser, Default, Debug)]
+pub struct MigrateSafeModuleArgs {
+    #[clap(help = "Network name. E.g. monte_rosa", long)]
+    network: String,
+
+    #[clap(flatten)]
+    local_identity: LocalIdentityArgs,
+
+    #[clap(flatten)]
+    password: PasswordArgs,
+
+    #[clap(
+        help = "Specify path pointing to the contracts root",
+        long,
+        short,
+        default_value = None
+    )]
+    contracts_root: Option<String>,
+
+    #[clap(
+        help = "Ethereum address of safe",
+        long,
+        short
+    )]
+    safe_address: String,
+
+    #[clap(
+        help = "Ethereum address of node management module",
+        long,
+        short
+    )]
+    module_address: String,
+
+    #[clap(
+        help = "Hopr amount in ether, e.g. 10",
+        long,
+        short = 't',
+        value_parser = clap::value_parser!(f64),
+        default_value_t = 2000.0
+    )]
+    hopr_amount: f64,
+
+    #[clap(
+        help = "Native token amount in ether, e.g. 1",
+        long,
+        short = 'n',
+        value_parser = clap::value_parser!(f64),
+        default_value_t = 10.0
+    )]
+    native_amount: f64,
+}
+
+impl MigrateSafeModuleArgs {
+    /// 1. Scope channels, announcement contracts
+    /// 2. Set approval for channels
+    /// 3. Include node to network registry
+    fn execute_safe_module_migration(self) -> Result<(), HelperErrors> {
+        let MigrateSafeModuleArgs {
+            network,
+            local_identity,
+            password,
+            contracts_root,
+            safe_address,
+            module_address,
+            hopr_amount,
+            native_amount,
+        } = self;
+
+        // 1. `PRIVATE_KEY` - Private key is required to send on-chain transactions
+        if let Err(_) = env::var("PRIVATE_KEY") {
+            return Err(HelperErrors::UnableToReadPrivateKey);
+        }
+
+        // 2. Calculate addresses from the identity file
+        // collect all the peer ids
+        let all_node_addresses: Vec<String>;
```

Comment:
Can we extract this step 2, into a separate function which I probably think that might be used by other hopli actions?

Reply:
Agree. Will do it when improving hopli with other improvements, such as calling directly forge